### PR TITLE
Split dev/production deployments

### PR DESCRIPTION
**Pull request by ludeeus**

### CODE REVIEW

File: .github/workflows/worker.yaml [hunk 1]
```diff
@@ -20,9 +21,15 @@ jobs:
         with:
           node-version: ${{ env.NODE_VERSION }}
 
-      - name: Publish worker
+      - name: Set environment
+        id: environment
+        run: |
+          echo "::set-ouput name=environment::production"
```
Comment (ludeeus) [inline .github/workflows/worker.yaml:27]: set production only if branch is main

**Comment by ludeeus** [comment]
Another alternative to dual branch, is to use <https://github.com/netlify/cli/blob/main/docs/commands/deploy.md> in GH action.

Then on merge to main update staging, and create repo releases for production updates
Personally I like that approach better.

File: worker/wrangler.toml [hunk 2]
```diff
@@ -1,13 +1,39 @@
...
 ]
+vars = {WORKER_ENV = "production"}
+
+
+# For staging environment, use '-e staging'
+[env.staging]
+name = "analytics-v1-staging"
+zone_id = "0ba583492080d3db28c103574f1d19cf"
+route = "https://analytics-api-staging.home-assistant.io/v1"
+kv_namespaces = [
```
Comment (ludeeus) [inline worker/wrangler.toml:25]: staging and dev should be different

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -4,6 +4,7 @@ on:
   pull_request:
     branches:
       - main
+      - staging
```
Comment (balloob) [inline .github/workflows/test.yaml:7]: Should we call the branch `staging` or just `dev` like we do for our other repos?

File: worker/src/handlers/schedule.ts [hunk 4]
```diff
@@ -96,7 +96,12 @@ async function processQueue(sentry: Toucan): Promise<void> {
     core_analytics[timestampString] = queue_data;
 
     sentry.addBreadcrumb({ message: "Trigger Netlify build" });
-    const resp = await fetch(NETLIFY_BUILD_HOOK, { method: "POST" });
+    const resp = await fetch(
+      WORKER_ENV === "production"
+        ? NETLIFY_BUILD_HOOK
+        : NETLIFY_BUILD_HOOK_STAGING,
```
Comment (balloob) [inline worker/src/handlers/schedule.ts:102]: We should always read from the same environment variables. The env variables should have different values based on their environment.

Comment (ludeeus) [reply worker/src/handlers/schedule.ts:102]: can they have that?

File: .github/workflows/test.yaml (hunk 3 above)
Comment (ludeeus) [reply .github/workflows/test.yaml:7]: Maybe :shrug:

File: worker/src/handlers/schedule.ts (hunk 4 above)
Comment (ludeeus) [reply worker/src/handlers/schedule.ts:102]: This is for the worker... yes, they can have same name

File: .github/workflows/test.yaml (hunk 3 above)
Comment (balloob) [reply .github/workflows/test.yaml:7]: I prefer `dev`.

**Review by balloob — APPROVED** [review]

**Pull request merged**